### DISCUSSION TITLE
Add support for CosmosDB connectivity mode

### DISCUSSION THREAD

**Discussion by escouser** [Azure/data-api-builder#1398] [Ideas]
Hi

I would like to request adding support for changing the connectivity mode when using CosmosDB.

CosmosDB has two connectivity modes, see https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/sdk-connection-modes - _the gateway mode is suitable for people working within a corporate network_.

This can be done by using the [WithConnectionModeGateway ](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.fluent.cosmosclientbuilder.withconnectionmodegateway?view=azure-dotnet) method when creating the CosmosClientBuilder e.g.

`Client = new CosmosClientBuilder(_connectionString).WithContentResponseOnWrite(true).WithConnectionModeGateway().Build();`

**Comment by sajeetharan** [comment]
Hi @escouser Thanks for sharing the feedback.

This is automatically managed by the Data API Builder Engine. We use the Direct Connectivity Mode since the engine leverages .Net SDK. Can you elaborate more on the requirement here?

**Reply by escouser** [reply]
Hi @sajeetharan 

So when developing locally, due to work firewall restrictions, I can only access CosmosDB using the Gateway mode (via HTTPS), it doesn't work using Direct, so I am unable to use the Data API Builder as it can't connect to CosmosDB.

I'm not sure what you mean by automatically managed, the change would be a way to call the WithConnectionModeGateway as part of this file https://github.com/Azure/data-api-builder/blob/main/src/Service/Resolvers/CosmosClientProvider.cs, lines 66, 70 and 75 - I've done this locally but unsure of how it should be configurable so it can use either Direct or Gateway.

Thanks

**Reply by sajeetharan** [reply]
Ok got it, I think you'd want to use the engine locally and you want to enable Gateway mode, let me get back on this!
